diff --git a/src/http/ngx_http_header_filter_module.c b/src/http/ngx_http_header_filter_module.c
index 76f6e9629..8cec934f9 100644
--- a/src/http/ngx_http_header_filter_module.c
+++ b/src/http/ngx_http_header_filter_module.c
@@ -46,9 +46,9 @@ ngx_module_t  ngx_http_header_filter_module = {
 };
 
 
-static u_char ngx_http_server_string[] = "Server: nginx" CRLF;
-static u_char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
-static u_char ngx_http_server_build_string[] = "Server: " NGINX_VER_BUILD CRLF;
+// static u_char ngx_http_server_string[] = "Server: nginx" CRLF;
+// static u_char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
+// static u_char ngx_http_server_build_string[] = "Server: " NGINX_VER_BUILD CRLF;
 
 
 static ngx_str_t ngx_http_status_lines[] = {
@@ -283,17 +283,17 @@ ngx_http_header_filter(ngx_http_request_t *r)
 
     clcf = ngx_http_get_module_loc_conf(r, ngx_http_core_module);
 
-    if (r->headers_out.server == NULL) {
-        if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
-            len += sizeof(ngx_http_server_full_string) - 1;
+    // if (r->headers_out.server == NULL) {
+    //     if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
+    //         len += sizeof(ngx_http_server_full_string) - 1;
 
-        } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
-            len += sizeof(ngx_http_server_build_string) - 1;
+    //     } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
+    //         len += sizeof(ngx_http_server_build_string) - 1;
 
-        } else {
-            len += sizeof(ngx_http_server_string) - 1;
-        }
-    }
+    //     } else {
+    //         len += sizeof(ngx_http_server_string) - 1;
+    //     }
+    // }
 
     if (r->headers_out.date == NULL) {
         len += sizeof("Date: Mon, 28 Sep 1970 06:00:00 GMT" CRLF) - 1;
@@ -379,27 +379,27 @@ ngx_http_header_filter(ngx_http_request_t *r)
         len += sizeof("Transfer-Encoding: chunked" CRLF) - 1;
     }
 
-    if (r->headers_out.status == NGX_HTTP_SWITCHING_PROTOCOLS) {
-        len += sizeof("Connection: upgrade" CRLF) - 1;
+    // if (r->headers_out.status == NGX_HTTP_SWITCHING_PROTOCOLS) {
+    //     len += sizeof("Connection: upgrade" CRLF) - 1;
 
-    } else if (r->keepalive) {
-        len += sizeof("Connection: keep-alive" CRLF) - 1;
+    // } else if (r->keepalive) {
+    //     len += sizeof("Connection: keep-alive" CRLF) - 1;
 
-        /*
-         * MSIE and Opera ignore the "Keep-Alive: timeout=<N>" header.
-         * MSIE keeps the connection alive for about 60-65 seconds.
-         * Opera keeps the connection alive very long.
-         * Mozilla keeps the connection alive for N plus about 1-10 seconds.
-         * Konqueror keeps the connection alive for about N seconds.
-         */
+    //     /*
+    //      * MSIE and Opera ignore the "Keep-Alive: timeout=<N>" header.
+    //      * MSIE keeps the connection alive for about 60-65 seconds.
+    //      * Opera keeps the connection alive very long.
+    //      * Mozilla keeps the connection alive for N plus about 1-10 seconds.
+    //      * Konqueror keeps the connection alive for about N seconds.
+    //      */
 
-        if (clcf->keepalive_header) {
-            len += sizeof("Keep-Alive: timeout=") - 1 + NGX_TIME_T_LEN + 2;
-        }
+    //     if (clcf->keepalive_header) {
+    //         len += sizeof("Keep-Alive: timeout=") - 1 + NGX_TIME_T_LEN + 2;
+    //     }
 
-    } else {
-        len += sizeof("Connection: close" CRLF) - 1;
-    }
+    // } else {
+    //     len += sizeof("Connection: close" CRLF) - 1;
+    // }
 
 #if (NGX_HTTP_GZIP)
     if (r->gzip_vary) {
@@ -452,22 +452,22 @@ ngx_http_header_filter(ngx_http_request_t *r)
     }
     *b->last++ = CR; *b->last++ = LF;
 
-    if (r->headers_out.server == NULL) {
-        if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
-            p = ngx_http_server_full_string;
-            len = sizeof(ngx_http_server_full_string) - 1;
+    // if (r->headers_out.server == NULL) {
+    //     if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
+    //         p = ngx_http_server_full_string;
+    //         len = sizeof(ngx_http_server_full_string) - 1;
 
-        } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
-            p = ngx_http_server_build_string;
-            len = sizeof(ngx_http_server_build_string) - 1;
+    //     } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
+    //         p = ngx_http_server_build_string;
+    //         len = sizeof(ngx_http_server_build_string) - 1;
 
-        } else {
-            p = ngx_http_server_string;
-            len = sizeof(ngx_http_server_string) - 1;
-        }
+    //     } else {
+    //         p = ngx_http_server_string;
+    //         len = sizeof(ngx_http_server_string) - 1;
+    //     }
 
-        b->last = ngx_cpymem(b->last, p, len);
-    }
+    //     b->last = ngx_cpymem(b->last, p, len);
+    // }
 
     if (r->headers_out.date == NULL) {
         b->last = ngx_cpymem(b->last, "Date: ", sizeof("Date: ") - 1);
@@ -555,23 +555,23 @@ ngx_http_header_filter(ngx_http_request_t *r)
                              sizeof("Transfer-Encoding: chunked" CRLF) - 1);
     }
 
-    if (r->headers_out.status == NGX_HTTP_SWITCHING_PROTOCOLS) {
-        b->last = ngx_cpymem(b->last, "Connection: upgrade" CRLF,
-                             sizeof("Connection: upgrade" CRLF) - 1);
+    // if (r->headers_out.status == NGX_HTTP_SWITCHING_PROTOCOLS) {
+    //     b->last = ngx_cpymem(b->last, "Connection: upgrade" CRLF,
+    //                          sizeof("Connection: upgrade" CRLF) - 1);
 
-    } else if (r->keepalive) {
-        b->last = ngx_cpymem(b->last, "Connection: keep-alive" CRLF,
-                             sizeof("Connection: keep-alive" CRLF) - 1);
+    // } else if (r->keepalive) {
+    //     b->last = ngx_cpymem(b->last, "Connection: keep-alive" CRLF,
+    //                          sizeof("Connection: keep-alive" CRLF) - 1);
 
-        if (clcf->keepalive_header) {
-            b->last = ngx_sprintf(b->last, "Keep-Alive: timeout=%T" CRLF,
-                                  clcf->keepalive_header);
-        }
+    //     if (clcf->keepalive_header) {
+    //         b->last = ngx_sprintf(b->last, "Keep-Alive: timeout=%T" CRLF,
+    //                               clcf->keepalive_header);
+    //     }
 
-    } else {
-        b->last = ngx_cpymem(b->last, "Connection: close" CRLF,
-                             sizeof("Connection: close" CRLF) - 1);
-    }
+    // } else {
+    //     b->last = ngx_cpymem(b->last, "Connection: close" CRLF,
+    //                          sizeof("Connection: close" CRLF) - 1);
+    // }
 
 #if (NGX_HTTP_GZIP)
     if (r->gzip_vary) {
