diff --git a/src/http/v2/ngx_http_v2_filter_module.c b/src/http/v2/ngx_http_v2_filter_module.c
index 9ffb155df..8d5d9b029 100644
--- a/src/http/v2/ngx_http_v2_filter_module.c
+++ b/src/http/v2/ngx_http_v2_filter_module.c
@@ -259,16 +259,18 @@ ngx_http_v2_header_filter(ngx_http_request_t *r)
 
     clcf = ngx_http_get_module_loc_conf(r, ngx_http_core_module);
 
-    if (r->headers_out.server == NULL) {
+    if (0) {
+        if (r->headers_out.server == NULL) {
 
-        if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
-            len += 1 + nginx_ver_len;
+            if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
+                len += 1 + nginx_ver_len;
 
-        } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
-            len += 1 + nginx_ver_build_len;
+            } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
+                len += 1 + nginx_ver_build_len;
 
-        } else {
-            len += 1 + sizeof(nginx);
+            } else {
+                len += 1 + sizeof(nginx);
+            }
         }
     }
 
@@ -463,46 +465,48 @@ ngx_http_v2_header_filter(ngx_http_request_t *r)
         pos = ngx_sprintf(pos, "%03ui", r->headers_out.status);
     }
 
-    if (r->headers_out.server == NULL) {
+    if(0) {
+        if (r->headers_out.server == NULL) {
 
-        if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
-            ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,
-                           "http2 output header: \"server: %s\"",
-                           NGINX_VER);
+            if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
+                ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,
+                            "http2 output header: \"server: %s\"",
+                            NGINX_VER);
 
-        } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
-            ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,
-                           "http2 output header: \"server: %s\"",
-                           NGINX_VER_BUILD);
+            } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
+                ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,
+                            "http2 output header: \"server: %s\"",
+                            NGINX_VER_BUILD);
 
-        } else {
-            ngx_log_debug0(NGX_LOG_DEBUG_HTTP, fc->log, 0,
-                           "http2 output header: \"server: nginx\"");
-        }
+            } else {
+                ngx_log_debug0(NGX_LOG_DEBUG_HTTP, fc->log, 0,
+                            "http2 output header: \"server: nginx\"");
+            }
 
-        *pos++ = ngx_http_v2_inc_indexed(NGX_HTTP_V2_SERVER_INDEX);
+            *pos++ = ngx_http_v2_inc_indexed(NGX_HTTP_V2_SERVER_INDEX);
 
-        if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
-            if (nginx_ver[0] == '\0') {
-                p = ngx_http_v2_write_value(nginx_ver, (u_char *) NGINX_VER,
-                                            sizeof(NGINX_VER) - 1, tmp);
-                nginx_ver_len = p - nginx_ver;
-            }
+            if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_ON) {
+                if (nginx_ver[0] == '\0') {
+                    p = ngx_http_v2_write_value(nginx_ver, (u_char *) NGINX_VER,
+                                                sizeof(NGINX_VER) - 1, tmp);
+                    nginx_ver_len = p - nginx_ver;
+                }
 
-            pos = ngx_cpymem(pos, nginx_ver, nginx_ver_len);
+                pos = ngx_cpymem(pos, nginx_ver, nginx_ver_len);
 
-        } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
-            if (nginx_ver_build[0] == '\0') {
-                p = ngx_http_v2_write_value(nginx_ver_build,
-                                            (u_char *) NGINX_VER_BUILD,
-                                            sizeof(NGINX_VER_BUILD) - 1, tmp);
-                nginx_ver_build_len = p - nginx_ver_build;
-            }
+            } else if (clcf->server_tokens == NGX_HTTP_SERVER_TOKENS_BUILD) {
+                if (nginx_ver_build[0] == '\0') {
+                    p = ngx_http_v2_write_value(nginx_ver_build,
+                                                (u_char *) NGINX_VER_BUILD,
+                                                sizeof(NGINX_VER_BUILD) - 1, tmp);
+                    nginx_ver_build_len = p - nginx_ver_build;
+                }
 
-            pos = ngx_cpymem(pos, nginx_ver_build, nginx_ver_build_len);
+                pos = ngx_cpymem(pos, nginx_ver_build, nginx_ver_build_len);
 
-        } else {
-            pos = ngx_cpymem(pos, nginx, sizeof(nginx));
+            } else {
+                pos = ngx_cpymem(pos, nginx, sizeof(nginx));
+            }
         }
     }
 
